tests/system/src/test/java/test/robot/javafx/scene/RTLTextCharacterIndexTest.java

```diff
@@ -0,0 +1,383 @@
+import javafx.scene.robot.Robot;
+import javafx.scene.text.Font;
+import javafx.scene.text.HitInfo;
+import javafx.scene.text.Text;
+import javafx.stage.Stage;
+import javafx.stage.StageStyle;
+import javafx.stage.Window;
+import org.junit.jupiter.api.AfterAll;
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import test.util.Util;
+
+/*
+ * Test for verifying character index of Text nodes in RTL orientation.
+ *
+ * There are 6 tests in this file.
+ * Here, the scene node orientation is set to RTL for all the tests.
+ * Steps for testTextInfoForRTLEnglishText()
+ * 1. Create a Text node and add it to the scene using a VBox.
+ * 2. Add only english text to the Text node.
+ * 3. Move the cursor from right to left with a random
+ *    decrement value generated in step() method.
+ * 4. Character index should change from highest value to lowest
+ *    as expected.
+ *
+ * Steps for testTextInfoForRTLArabicText()
+ * 1. Create a Text node and add it to the scene using a VBox.
+ * 2. Add only arabic text to the Text node.
+ * 3. Move the cursor from right to left with a random
+ *    decrement value generated in step() method.
+ * 3. Character index should increment as expected since it is RTL text.
+ *
+ * Steps for testTextInfoForRTLEnglishArabicText()
+ * 1. Create a Text node and add it to the scene using a VBox.
+ * 2. Add both english and arabic text to the Text node.
+ * 3. Move the cursor from right to left with a random
+ *    decrement value generated in step() method.
+ * 4. Character index should change in decreasing order for english text
+ *    and in increasing order for arabic text.
+ *
+ * Steps for testTextInfoForMultiLineRTLEnglishText()
+ * 1. Create a Text node and add it to the scene using a VBox.
+ * 2. Add two lines of only english text to the Text node.
+ * 3. Move the cursor from right to left with a random
+ *    decrement value generated in step() method.
+ * 4. Character index should change in decreasing order as expected.
+ *
+ * Steps for testTextInfoForMultiLineRTLEnglishArabicText()
+ * 1. Create a Text node and add it to the scene using a VBox.
+ * 2. Add two lines of both english and arabic text to the Text node.
+ * 3. Move the cursor from right to left with a random
+ *    decrement value generated in step() method.
+ * 4. Character index should change in decreasing order for english text
+ *    and increasing order for arabic text.
+ *
+ * Steps for testTextInfoForMultiLineRTLArabicText()
+ * 1. Create a Text node and add it to the scene using a VBox.
+ * 2. Add two lines of only arabic text to the Text node.
+ * 3. Move the cursor from right to left with a random
+ *    decrement value generated in step() method.
+ * 4. Character index should change in increasing order as expected.
+ */
+
+public class RTLTextCharacterIndexTest {
```

Comment:
This is a system test, which don't run with the build system. Are you sure this will work on all platforms?  I don't see a specific `Font` used, which means `X_LEADING_OFFSET` may be incorrect when the platform is supplying a different font.

Reply:
I see that the tests doesn't run on all the platforms. I will look into this and update the PR

Reply:
A valid concern about specific fonts, similar to 
https://github.com/openjdk/jfx/pull/1236#issuecomment-1937022815

What was the resolution there?  To use a hard-coded font based on the platform, or pick an available one from the list?

Reply:
The tests passes on Mac and Linux. The  failures I mentioned above were because this branch was not upto date with the master and completely unrelated to this PR.
Im facing some issue with my Windows system. I will check in windows once it is sorted and update the details here.

Reply:
I have run the tests in Windows and did not see any failures.